### DISCUSSION TITLE
Customizing UI

### DISCUSSION THREAD

**Discussion by leniervm** [apache/streampipes#3039] [Q&A] [answered]
Hi!, is there a way to customize UI using docker compose file to run StreamPipes?

**Comment by dominikriemer** [comment]
Hi @leniervm you can create your own customized UI build, a manual can be found here: https://streampipes.apache.org/docs/extend-customize-ui/

**Comment by leniervm** [comment]
Thank you, @dominikriemer. As a professional transitioning from OT to IT, I have some experience working with some applications in the Laravel Framework. I understand that I need to edit the customizable files in the UI folder, install npm dependencies, and then build the project. After these steps, I will need to pull a new Docker image with my updated UI to integrate it into my project. Is this correct?

**Comment by dominikriemer** [comment, marked as answer]
@leniervm that's correct. You don't need to edit the customizable files, but can have a separate folder with the logo, colors and application title. You can then trigger the build by providing environment variables which link to the customized files.

While this approach seems complicated for "just" changing colors and texts, it has the advantage that you can create builds with customized UI views (e.g., if you don't need the live dashboard, you can create a StreamPipes instance without the dashboard view) and you can even create your own user interfaces without touching the StreamPipes code using a microfrontend approach - just like a plugin system. See some example plugins which my company provides on top of StreamPipes, just for inspiration: https://bytefabrik.ai/en/product-manufacturing-insights/

We might provide an easier way for just changing colors in the future, but it's not yet planned.

**Reply by leniervm** [reply]
It must be done with Streampipes CLI or it can be executed in docker compose based installation @dominikriemer?
